src/routes/products.routes.ts

```diff
@@ -24,4 +24,8 @@ productRoutes.patch(
   productsController.updateProduct,
 );
 
+productRoutes.get('/product/:product_id',
+isAuthorized(UserRole.ADMIN,UserRole.VENDOR,UserRole.USER),
```

Comment:
what's the difference between using this where you've mentioned every role and using `isAuthenticated`?

Reply:
> what's the difference between using this where you've mentioned every role and using `isAuthenticated`?

there were issues with the middleware for getting req.user and i chose to use mine that get user and checks the roles as well, i have mentioned every role as the route must be accessible for all roles.